test/metabase/email_test.clj

```diff
@@ -359,3 +359,36 @@
               (is (re-find
                    #"(?s)Content-Disposition: attachment.+filename=.+this-is-quite-[\-\s?=0-9a-zA-Z]+-characters.csv"
                    (m/mapply email/send-message! params-with-problematic-file))))))))))
+
+(deftest throttle-test
```

Comment:
It would be great to have tests for the following:

1. Two small events that don't get throttled, then a third small one that does.
1. A small event then a large event, where the second event crosses the threshold.[^1]
3. The limit resetting and the next event being accepted.

Not sure if throttle has anything nice to advance its clock without actually sleeping, otherwise perhaps we can swap it out with one with a really short timeout.

[^1]: IMHO this second one should immediately be blocked, although alternatively we could allow surging instead past instead and only throttle a third large event coming afterwards.

Reply:
does this capture it? : 25e0371da5

Reply:
Nailed it 💅